src/runtimeHelpers.ts

```diff
@@ -62,38 +62,81 @@ export const cacheKeyOptions = {
 
 /**
  * Take in all objects passed to .load(), and bucket them by the non
- * batchKey attributes.
+ * batchKeys(including batchKey and secondaryBatchKey) attributes.
  *
  * We use this to chunk up the requests to the resource.
  *
  * Example:
  * ```js
- * partitionItems([
+ * partitionItems('bar_id', [
  *   { bar_id: 2, include_extra_info: true },
  *   { bar_id: 3, include_extra_info: false },
  *   { bar_id: 4, include_extra_info: true },
- * ], 'bar_id')
+ * ])
  * ```
  *
  * Returns:
  * `[ [ 0, 2 ], [ 1 ] ]`
  *
  * TODO: add generic instead of 'object' for the items array argument
  */
-export function partitionItems(ignoreKey: string, items: ReadonlyArray<object>): ReadonlyArray<ReadonlyArray<number>> {
+export function partitionItems(
+    ignoreKeys: Array<string> | string,
+    items: ReadonlyArray<object>,
+): ReadonlyArray<ReadonlyArray<number>> {
     const groups: {
         [key: string]: Array<number>;
     } = {};
-
     items.forEach((item, i) => {
-        const hash = objectHash(_.omit(item, ignoreKey), { algorithm: 'passthrough' });
+        const hash = objectHash(_.omit(item, ignoreKeys), { algorithm: 'passthrough' });
         groups[hash] = groups[hash] || [];
         groups[hash].push(i);
     });
 
     return Object.values(groups);
 }
 
+/**
+ * Take in all objects passed to .load(), and bucket them by the non
+ * batchKeys(including batchKey and secondaryBatchKey) attributes.
+ * Return batchKey value for each partition items.
+ *
+ * This function is only called when we have secondaryBatchKey, and it's
+ * used to map result to the order of requests.
+ *
+ * Example:
+ * ```js
+ * getBatchKeyForPartitionItems(
+ *    'bar_id',
+ *    ['bar_id', 'property'],
+ *    [
+ *      { bar_id: 2, property: 'name', include_extra_info: true },
+ *      { bar_id: 3, property: 'rating', include_extra_info: false },
+ *      { bar_id: 2, property: 'rating', include_extra_info: true },
+ *    ])
+ * ```
+ *
+ * Returns:
+ * `[ [ 2, 2 ], [ 3 ] ]`
+ *
+ * TODO: add generic instead of 'object' for the items array argument
+ */
+export function getBatchKeyForPartitionItems(
+    batchKey: string,
+    ignoreKeys: Array<string>,
+    items: ReadonlyArray<object>,
+): ReadonlyArray<ReadonlyArray<number>> {
```

Comment:
Could `partitionItems` be re-used? What is this doing differently?

Reply:
instead of returning the order, we return the batchKey value of that item, the main difference is line 135. https://github.com/Yelp/dataloader-codegen/pull/261/files#diff-92b6f52e2c1b302cd9d15133831c595f151215aa1dfda274ff29fbd33d89a9b2R135 

For example, if we have (the batch key here is `bar_id` and secondary batch key is `property`) : 
```
    { bar_id: 100, property: 'name', include_extra_info: true },
    { bar_id: 101, property: 'rating', include_extra_info: false },
    { bar_id: 100, property: 'rating', include_extra_info: true },
 ```
`partitionItems` will return [ [ 0, 2 ], [ 1 ] ]
while, `getBatchKeyForPartitionItems ` with return [ [ 100, 100 ], [ 101 ] ]
We use this information to help us match results object to requests. see https://github.com/Yelp/dataloader-codegen/pull/261/files#diff-92b6f52e2c1b302cd9d15133831c595f151215aa1dfda274ff29fbd33d89a9b2R349-R350